app/code/Magento/MediaContentSynchronization/Model/SynchronizeIdentities.php

```diff
@@ -0,0 +1,120 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MediaContentSynchronization\Model;
+
+use Magento\Framework\App\ResourceConnection;
+use Magento\MediaContentApi\Api\Data\ContentIdentityInterfaceFactory;
+use Magento\MediaContentApi\Api\UpdateContentAssetLinksInterface;
+use Magento\MediaContentApi\Model\GetEntityContentsInterface;
+use Magento\MediaContentSynchronizationApi\Api\SynchronizeIdentitiesInterface;
+
+class SynchronizeIdentities implements SynchronizeIdentitiesInterface
+{
+    private const ENTITY_TYPE = 'entityType';
+    private const ENTITY_ID = 'entityId';
+    private const FIELD = 'field';
+
+    private const FIELD_CMS_PAGE = 'cms_page';
+    private const FIELD_CMS_BLOCK = 'cms_block';
+
+    private const ID_CMS_PAGE = 'page_id';
+    private const ID_CMS_BLOCK = 'block_id';
+
+    private const COLUMN_CMS_CONTENT = 'content';
+
+    /**
+     * @var ResourceConnection
+     */
+    private $resourceConnection;
+
+    /**
+     * @var ContentIdentityInterfaceFactory
+     */
+    private $contentIdentityFactory;
+
+    /**
+     * @var UpdateContentAssetLinksInterface
+     */
+    private $updateContentAssetLinks;
+
+    /**
+     * @var GetEntityContentsInterface
+     */
+    private $getEntityContents;
+
+    /**
+     * @param ResourceConnection $resourceConnection
+     * @param ContentIdentityInterfaceFactory $contentIdentityFactory
+     * @param UpdateContentAssetLinksInterface $updateContentAssetLinks
+     * @param GetEntityContentsInterface $getEntityContents
+     */
+    public function __construct(
+        ResourceConnection $resourceConnection,
+        ContentIdentityInterfaceFactory $contentIdentityFactory,
+        UpdateContentAssetLinksInterface $updateContentAssetLinks,
+        GetEntityContentsInterface $getEntityContents
+    ) {
+        $this->resourceConnection = $resourceConnection;
+        $this->contentIdentityFactory = $contentIdentityFactory;
+        $this->updateContentAssetLinks = $updateContentAssetLinks;
+        $this->getEntityContents = $getEntityContents;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function execute(array $mediaContentIdentities): void
+    {
+        foreach ($mediaContentIdentities as $identity) {
+            $contentIdentity = $this->contentIdentityFactory->create(
+                [
+                    self::ENTITY_TYPE => $identity[self::ENTITY_TYPE],
+                    self::ENTITY_ID => $identity[self::ENTITY_ID],
+                    self::FIELD => $identity[self::FIELD]
+                ]
+            );
+
+            if ($identity[self::ENTITY_TYPE] === self::FIELD_CMS_PAGE
+                || $identity[self::ENTITY_TYPE] === self::FIELD_CMS_BLOCK
+            ) {
+                $content = $this->getCmsMediaContent(
```

Comment:
Please introduce the implementation of SynchronizeIdentitiesInterface to MediaContentCms and MediaContentCatalog modules and add them to this class via DI configuration (similar to \Magento\MediaContentSynchronization\Model\Synchronize::execute implemenation)